src/env.h

```diff
@@ -455,6 +455,31 @@ class Environment {
     DISALLOW_COPY_AND_ASSIGN(AsyncCallbackScope);
   };
 
+  class ImmediateInfo {
+   public:
+    inline uint32_t* fields();
+    inline int fields_count() const;
+    inline uint32_t count() const;
+    inline uint32_t outstanding() const;
+
+    inline void count_inc(uint32_t increment);
+    inline void count_dec(uint32_t decrement);
+
+   private:
+    friend class Environment;  // So we can call the constructor.
+    inline ImmediateInfo();
+
+    enum Fields {
+      kCount,
+      kOutstanding,
+      kFieldsCount
+    };
+
+    uint32_t fields_[kFieldsCount];
```

Comment:
From an initial look: It’s your call whether turning this into a class is helpful or not, but in any case the storage should only be accessed as an `AliasedBuffer`, since that is necessary for ChakraCore’s time-travel debugging to work with these values (because otherwise it can’t witness the changes from C++).

If you miss the availability of `++` and `--`, I guess it’s fair to add those to the `AliasedBuffer` reference class?

Reply:
Thanks, wasn't aware of that. Will make some changes.